pkg/systemd/hw-detect.es6

```diff
@@ -54,8 +54,22 @@ function findPCI(udevdb, info) {
     }
 }
 
+// Add to info.memory.dimm = [{"speed":"11 MHz","locator":"A3", ...}]
+function processMemory(memory, info) {
+    for (let dimm in memory) {
+        let props = memory[dimm];
+        info.memory.push({ locator: props["Locator"],
+                           manufacturer: props["Manufacturer"],
+                           type_detail: props["Type Detail"],
+                           size: props["Size"],
+                           speed: props["Speed"],
+                           part_number: props["Part Number"],
+                           serial: props["Serial Number"] });
+    }
+}
+
```

Comment:
It feels like this should be part of `parseMemoryInfo()`, as this is all about parsing the program output, and not related to UI presentation. If `dmidecode` changes output, you should only have to change one place, and keep the `memory_info()` API stable.